micro_sam/training/sam_trainer.py

```diff
@@ -117,6 +118,24 @@ def _compute_iou(self, pred, true, eps=1e-7):
         iou = overlap / (union + eps)
         return iou
 
+    def preprocess_one_hot_masks(self, y_one_hot):
+        """Converts the masks to match the "low_res_masks" shape.
+        """
+        # Convert the labels to "low_res_mask" shape
+        # First step is to use the logic from `ResizeLongestSide` to resize the longest side.
+        target_length = self.model.transform.target_length
+        target_shape = self.model.transform.get_preprocess_shape(y_one_hot.shape[2], y_one_hot.shape[3], target_length)
+        y_one_hot = F.interpolate(input=y_one_hot, size=target_shape)
+        # Next, we pad the remaining region to (1024, 1024)
+        h, w = y_one_hot.shape[-2:]
+        padh = self.model.sam.image_encoder.img_size - h
+        padw = self.model.sam.image_encoder.img_size - w
+        y_one_hot = F.pad(input=y_one_hot, pad=(0, padw, 0, padh))
```

Comment:
Is this not just `ResizeLongsetSide` until here? Can't we just use it instead of duplicating the code?

Reply:
It's because of one parameter that I have to remplement this: (they set antialias to True, which for labels here, we need to set to False.) https://github.com/facebookresearch/segment-anything/blob/main/segment_anything/utils/transforms.py#L63-L65

Reply:
Also to add: with `ResizeLongestSide`, it's still a two stage agenda: (taking LIVECell (images of shape `(520, 704)` as an example)

- `ResizeLongestSide` would only adjust the inputs to the longest side: (eg. `(520, 704)` -? `(720, 1024)`).
    - https://github.com/facebookresearch/segment-anything/blob/main/segment_anything/utils/transforms.py#L55-L65
- There's still the next stage where the padding takes place to complete the image shape to the desired input shape: (eg. `(720, 1024)` -> `(1024, 1024)`)
    - This happens here: https://github.com/facebookresearch/segment-anything/blob/main/segment_anything/modeling/sam.py#L164-L174

Reply:
I see, makes sense!